governance-aepp/src/utils/aeternity.js

```diff
@@ -89,6 +90,33 @@ aeternity.initStaticClient = async () => {
   });
 };
 
+aeternity.verifyPollContract = async (pollAddress) => {
+  const contractCreateBytecode = await fetch(`${settings[aeternity.networkId].middlewareUrl}/middleware/contracts/transactions/address/${pollAddress}?limit=1&page=1`).then(async res => {
```

Comment:
do we have to rely on the middleware for this?

Reply:
yes, we have to in order to fetch the latest bytecode, if middleware doesn't work reliable we can still weaken the error message in the frontend.